src/interfaces/IERC4973.sol

```diff
@@ -75,16 +75,21 @@ interface IERC4973 {
   ///  exists as an `uint256 tokenId` in the contract, `function take(...)` must
   ///  throw.
   /// @param from The origin of the ABT.
-  /// @param uri A distinct Uniform Resource Identifier (URI) for a given ABT.
+  /// @param metadata The metadata that will be associated to the ABT.
   /// @param signature A EIP-2098-compatible Compact Signature of the EIP-712
   ///  structured data hash
-  ///  `Agreement(address active,address passive,string tokenURI)` signed by
+  ///  `Agreement(address active,address passive,bytes metadata)` signed by
   ///  `address from`.
   /// @return A unique `uint256 tokenId` generated by type-casting the `bytes32`
   ///  EIP-712 structured data hash to a `uint256`.
   function take(
     address from,
-    string calldata uri,
+    bytes calldata metadata,
     bytes calldata signature
   ) external returns (uint256);
+  /// @notice Decodes the opaque metadata bytestring of an ABT into the token
+  /// URI that will be associated with it once it is created on chain.
+  /// @param metadata The metadata that will be associated to an ABT.
+  /// @return A URI that represents the metadata.
+  function decodeURI(bytes calldata metadata) external returns (string memory);
```

Comment:
Why are we including decodeURI in the interface? Shouldn‘t the interface user mainly interact with function tokenURI?

Reply:
The problem is that `tokenURI` is only available after on-chain minting. The `metadata` value represents a URI but in an opaque way, and a wallet that holds that value wouldn't be able to  decode it into a URI, so the decoding function should be a part of the interface. You want to know the URI before you mint it.

Reply:
reasonable point, I had never thought about it this way.